app/src/main/java/com/github/drawyourpath/bootcamp/authentication/GoogleAuthBootcampActivity.kt

```diff
@@ -0,0 +1,58 @@
+package com.github.drawyourpath.bootcamp.authentication
+
+import android.os.Bundle
+import android.widget.Button
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.github.drawyourpath.bootcamp.R
+import com.github.drawyourpath.bootcamp.authentication.provider.AuthProvider
+import com.github.drawyourpath.bootcamp.authentication.provider.GoogleSignInAuthProvider
+import com.github.drawyourpath.bootcamp.authentication.provider.MockAuthProvider
+
+const val GA_USE_MOCK_AUTH_PROVIDER_KEY = "useMockAuthProvider";
+
+class GoogleAuthBootcampActivity() : AppCompatActivity() {
+
+    private var authProvider: AuthProvider = GoogleSignInAuthProvider(this);
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_bootcamp_google_auth)
+
+        val signInButton = findViewById<Button>(R.id.bc_google_signin)
+        val signOutButton = findViewById<Button>(R.id.bc_google_signout);
+
+        val useMockAuthProvider = intent.getBooleanExtra(GA_USE_MOCK_AUTH_PROVIDER_KEY, false)
+        if (useMockAuthProvider) {
+            authProvider = MockAuthProvider();
+        }
```

Comment:
Question: Is this part absolutely needed ? I guess it is used for testing but I wonder if there is a way to avoid having this "extra case".